controlplane/kubeadm/internal/cluster.go

```diff
@@ -183,15 +183,88 @@ func (m *ManagementCluster) RemoveMachineFromKubeadmConfigMap(ctx context.Contex
 		// Nothing to do, no node for Machine
 		return nil
 	}
+	c, err := m.getCluster(ctx, clusterKey)
+	if err != nil {
+		return err
+	}
 
-	cluster, err := m.getCluster(ctx, clusterKey)
+	configMapKey := types.NamespacedName{Name: "kubeadm-config", Namespace: metav1.NamespaceSystem}
```

Comment:
```suggestion
	configMapKey := client.ObjectKey{Name: "kubeadm-config", Namespace: metav1.NamespaceSystem}
```

Reply:
Are we standardizing on client.ObjectKey? Are you thinking about this in terms of things getting passed to the `client.Client` should be the client.ObjectKey? Is there some semantic reason to use one or the other?

Reply:
I believe we've been standardizing on it, if nothing else in many cases it eliminates a need for an additional import.

Reply:
sgtm

Reply:
There are a ton of types.NamespacedName all over the place. Mind if we track this with an issue?

Reply:
I suspect many of them can likely be replaced with `util.ObjectKey` as well, so I think an issue is probably warranted :joy:

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/2459

yeah, it's not a tough ticket, but it's a lot of line changes